### PR TITLE
Feat/ebay cog

### PR DESCRIPTION
**Pull request by issy**
This is a price checking cog that checks the ebay website for pricing on an item.
Command aliases are as follows:
- check
- pc
- chk
- price
- pricecheck
Words can be filtered out of the search by using the `-` operator. For example: `!check dell xps 15 -broken -repair`
The command removes outliers by selecting the 25th to 75th percentile and then returns the range, median, average and variance of that boxplot

### CODE REVIEW

File: cogs/ebay.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+                for word in title.split(' '):
+                    if word.lower() in filteredWords:
+                        return True
+                return False
+
+            filteredListings = [i for i in products if filtered_out(i['title'], filteredWords) is False]
+            if len(filteredListings) == 0:
+                desc = [f"No results found for `{filteredTerm}`"]
+                desc.append("Make sure to include manufacturer for best results")
+                desc.append("If you feel this is in error, message Issy")
```
Comment (bitshiftsdotio) [inline cogs/ebay.py:49]: you're not the only person that maintains (or can maintain) this bot...

File: cogs/ebay.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+                    products.append(p)
+
+            def filtered_out(title: str, filteredWords: list):
+                for word in title.split(' '):
+                    if word.lower() in filteredWords:
+                        return True
+                return False
+
+            filteredListings = [i for i in products if filtered_out(i['title'], filteredWords) is False]
+            if len(filteredListings) == 0:
```
Comment (bitshiftsdotio) [inline cogs/ebay.py:46]: for what it's worth, empty lists in Python are falsy, so you could rewrite this as:

```suggestion
            if not filteredListings:
```

File: cogs/ebay.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+            d['url'] = product.find('a')['href']
+            d['image'] = product.find('img')['src']
+            d['price'] = await self.parse_price(product)
+            d['ended_date'] = await self.parse_date(product)
+            d['id'] = d['url'].split('/')[-1].split('?')[0]
+            return {'title': str(d['title']), 'url': str(d['url']), 'image': str(d['image']), 'price': d['price'], 'ended_date': d['ended_date'], 'id': str(d['id'])}
+        except:
+            return None
+
+    async def parse_price(self, product):
```
Comment (bitshiftsdotio) [inline cogs/ebay.py:140]: you reckon you could comment this to make it easier to digest what's going on please?

**Review by bitshiftsdotio — CHANGES_REQUESTED** [review]
some general comments on top: your code could do with some formatting (empty lines to make it easier to read). also, you should not be using `camelCase` variable names in Python, PEP-8 calls the use of `snake_case` instead

**Comment by issy** [comment]
> some general comments on top: your code could do with some formatting (empty lines to make it easier to read). also, you should not be using `camelCase` variable names in Python, PEP-8 calls the use of `snake_case` instead

I'm aware of the PEP-8 naming conventions, and I disagree with them. I purposefully use camelCase for non-global variables and snake_case for functions to distinguish between the two. For me, this is much more readable and this ideology is applied to 90% of the code in this repo.

**Pull request merged**